code/modules/antagonists/heretic/magic/ash_jaunt.dm

```diff
@@ -20,6 +20,59 @@
 	jaunt_type = /obj/effect/dummy/phased_mob/spell_jaunt/red
 	jaunt_in_type = /obj/effect/temp_visual/dir_setting/ash_shift
 	jaunt_out_type = /obj/effect/temp_visual/dir_setting/ash_shift/out
+	/// If we are on fire while wearing ash robes, we can empower our next cast
+	var/empowered_cast = FALSE
+
+/datum/action/cooldown/spell/jaunt/ethereal_jaunt/ash/Grant(mob/grant_to)
+	. = ..()
+	RegisterSignal(grant_to, COMSIG_FIRE_STACKS_UPDATED, PROC_REF(update_status_on_signal))
+
+/datum/action/cooldown/spell/jaunt/ethereal_jaunt/ash/Remove(mob/remove_from)
+	. = ..()
+	UnregisterSignal(remove_from, COMSIG_FIRE_STACKS_UPDATED)
+
+/datum/action/cooldown/spell/jaunt/ethereal_jaunt/ash/apply_button_overlay(atom/movable/screen/movable/action_button/current_button, force)
+	. = ..()
+	// Put an active border whenever our spell is able to be casted empowered
+	if(!ishuman(owner))
+		return
+	var/mob/living/carbon/human/human_owner = owner
+	if(!istype(human_owner.wear_suit, /obj/item/clothing/suit/hooded/cultrobes/eldritch/ash))
+		return
+	if(human_owner.fire_stacks <= 3)
+		return
+
+	current_button.cut_overlay(current_button.button_overlay)
+	current_button.button_overlay = mutable_appearance(icon = overlay_icon, icon_state = "bg_spell_border_active_green")
+	current_button.add_overlay(current_button.button_overlay)
+
+/datum/action/cooldown/spell/jaunt/ethereal_jaunt/ash/cast(mob/living/cast_on)
+	if(!iscarbon(owner))
+		return ..()
+	var/mob/living/carbon/carbon_owner = owner
+	carbon_owner.uncuff()
+	var/obj/item/clothing/shoes/shoes = carbon_owner.shoes
+	if(istype(shoes) && shoes.tied == SHOES_KNOTTED)
+		shoes.adjust_laces(SHOES_TIED, carbon_owner)
```

Comment:
Did we talk about this at some point? I remember thinking this is super lame and powercreepy

Reply:
I don't think it needs any extra oomph, it's a jaunt